documentation/add/07-sqw_redesign.md

```diff
@@ -1,58 +1,84 @@
-- "DND" which contains only processed neutron image data
+The two objects are treated by Horace as 'first class citizens'. The objects share a common API where that is appropriate. In an sqw object, operations are performed on the pixel array data and the `dnd` image data recalculated from this. For `dnd` objects the operations are performed directly on the image data.
+In addition, the extra information and raw data contained on the `sqw` object allows a scattering model to be fitted.
 
-The two objects are treated by Horace as 'first class citizens'. The objects share a common API where that is appropriate. In an SQW object, operations are performed on the pixel array data and the current image data recalculated from this. For DND objects the operations are performed directly on the image data. 
-In addition, the extra information and raw data contained on the SQW object allows a scattering model to be fitted.
+## Main Classes:
 
-## Main Classes
+The class diagram describing the relationship between main Horace classes is provided on the Fig.1:
 
-![Class Overview](../diagrams/sqw-dnd.png)
+![Fig.1. Class Overview](../diagrams/sqw-dnd.png)
 
 
 
 ### SQWDnDBase
 
-Abstract base class for the SQW and DND objects.
+Abstract base class for the `sqw` and `dnd` objects, describing the interface common for both `sqw` and `dnd` objects.
 
-This includes the image data and main header information, available in both the `SQW` and `DnD` classes.
+May include code of the methods with work identically on both `sqw` and `dnd` objects. .
 
-This class will also include common methods, in particular the large number of unary and binary operations which are implemented as calls to operations managers which will be defined in the implementing classes.
+Currently it also include common methods, in particular the large number of unary and binary operations which are implemented as calls to operations managers which will be defined in the implementing classes.
 
 ### SQW
 
-The `SQW` object is the providing the public API to data. Data manipulations are performed on the `PixelData` and `Image` is recalculated.
+The `sqw` object provides the public API to the all relevant experimental data. Main data manipulations are performed on the `PixelData` and `Image` (`dnd` object, within the `sqw` object) is recalculated accordingly.
 
-This class includes the full experiment data including the raw pixel data and details of the instrument and detectors.
+This class includes the full experiment data including the raw pixel data and details of the instrument and detectors. As the `PixelData` containing all information about neutron events is normally very large dataset `sqw` object for some purpoese may be used without `PixelData`. Alternatively, when `PixelData` is large and can not be loaded in memory, the operations on the `PixelData` can be performed on separate data chunks loaded and processed in memory, leaving the main `PixelData` arrays filebased.
```

Comment:
instrument, SAMPLE and detectors - add sample
add 'a' before very large dataset
add 'the' after large dataset
correct typo - purposes
Does 'without pixeldata' mean that the data is just not touched, or that it is removed from the object

Reply:
hopefully accepted all except can not find position for the before large.